src/Microsoft.DotNet.Cli.Utils/BuiltInCommand.cs

```diff
@@ -52,6 +54,11 @@ public CommandResult Execute()
                     // Reset the Reporters to the new Console Out and Error.
                     Reporter.Reset();
 
+                    if (!string.IsNullOrEmpty(_workingDirectory))
+                    {
+                        Directory.SetCurrentDirectory(_workingDirectory);
```

Comment:
Personally, I would have created an additional "known" argument to `compile-csc`, like `temp-out`, that takes the working directory for `csc` as the argument value. This would avoid manipulating the current exe's working directory.